### ISSUE TITLE
`overly_complex_bool_expr` triggers wrongly on `(return) || true`

### ISSUE THREAD

**Issue by bend-n** [rust-lang/rust-clippy#12744]
### Summary

`_ = (return) || true` suggests being replaced with `true`

### Lint Name

overly_complex_bool_expr

### Reproducer

I tried this code:

```rust
#![allow(unreachable_code, clippy::never_loop, clippy::diverging_sub_expression)]
fn main() {
    _ = (return) || true;
}
```

I saw this happen:

```

error: this boolean expression contains a logic bug
 --> src/main.rs:3:9
  |
3 |     _ = (return) || true;
  |         ^^^^^^^^^^^^^^^^ help: it would look like the following: `true`
  |
help: this expression can be optimized out by applying boolean operations to the outer expression
 --> src/main.rs:3:9
  |
3 |     _ = (return) || true;
  |         ^^^^^^^^
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#overly_complex_bool_expr
  = note: `#[deny(clippy::overly_complex_bool_expr)]` on by defau
```

I expected to see this happen:

### Version

```text
rustc 1.80.0-nightly (c987ad527 2024-05-01)
binary: rustc
commit-hash: c987ad527540e8f1565f57c31204bde33f63df76
commit-date: 2024-05-01
host: x86_64-unknown-linux-gnu
release: 1.80.0-nightly
LLVM version: 18.1.4
```

### Additional Labels

@rustbot label +I-suggestion-causes-bug

**Comment by bend-n** [comment]
wait im an idiot
